src/db2ia/dbstmt.cc

```diff
@@ -2776,14 +2776,15 @@ int DbStmt::bindParams(Napi::Env env, Napi::Array *params, std::string &error)
           std::string string = value.ToString().Utf8Value();
           size_t str_length = string.length();
           const char *cString = string.c_str();
-          
-          if (cString[0] == '\0') // Check for JS empty-string.
-           {
-            param[i].ind = SQL_NTS;
-          } else {
-            param[i].ind = str_length;
-          }
-          param[i].buf = strdup(cString);
+          // CLI does not honor the buffer size parameter or the output size in the indicator
+          // Make the buffer size is at least the size of parameter or the size of the string
+          param[i].buf = (char *)calloc(std::max(static_cast<size_t>(param[i].paramSize) + 1, str_length), sizeof(char));
+          // set the indicator to be SQL_NTS
+          // this helps in edge cases like empty string where indicator is set 0
+          param[i].ind = SQL_NTS;
+          // SQL_PARAM_INPUT_OUTPUT is always set so
+          // copy the string to the buffer
+          strcpy((char*)param[i].buf, cString);

```

Comment:
Same changes as above need to be done here too

Reply:
Updated